### PR TITLE
Added missing APIs to complete RP-ARM SLA

### PR DESCRIPTION
**Pull request by akumardu**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/iothub/resource-manager/Microsoft.Devices/stable/2018-04-01/examples/iothub_testallroutes.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  "responses": {
+    "200": {
+      "body": {
+        "routes": [
+          {
+            "properties": {
+                "name": "Route id",
+                "source": "DeviceMessages",
+                "endpointNames": ["id1", "id2"],
+                "isEnabled": "true"
```
Comment (jianghaolu) [inline specification/iothub/resource-manager/Microsoft.Devices/stable/2018-04-01/examples/iothub_testallroutes.json]: This should be a boolean.

File: specification/iothub/resource-manager/Microsoft.Devices/stable/2018-04-01/iothub.json [hunk 2]
```diff
@@ -167,68 +183,76 @@
...
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Resource group identifier."
+          },
+          {
+            "name": "resourceName",
+            "in": "path",
+
```
Comment (jianghaolu) [inline specification/iothub/resource-manager/Microsoft.Devices/stable/2018-04-01/iothub.json]: Remove empty lines.

**Review by jianghaolu — CHANGES_REQUESTED** [review]
Couple minor issues - please take a look.

File: specification/iothub/resource-manager/Microsoft.Devices/stable/2018-04-01/examples/iothub_testallroutes.json (hunk 1 above)
Comment (akumardu) [reply specification/iothub/resource-manager/Microsoft.Devices/stable/2018-04-01/examples/iothub_testallroutes.json]: Ack.

File: specification/iothub/resource-manager/Microsoft.Devices/stable/2018-04-01/examples/iothub_testallroutes.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "routes": [
+          {
+            "properties": {
+                "name": "Route id",
+                "source": "DeviceMessages",
+                "endpointNames": ["id1", "id2"],
```
Comment (jianghaolu) [inline specification/iothub/resource-manager/Microsoft.Devices/stable/2018-04-01/examples/iothub_testallroutes.json]: Sorry for the late response. Here is another issue I found - this array is defined to have max length of 1: https://github.com/Azure/azure-rest-api-specs/pull/3031/files#diff-54dafbe4198f749bac837b913372bcddR2417

**Review by jianghaolu — CHANGES_REQUESTED** [review]

File: specification/iothub/resource-manager/Microsoft.Devices/stable/2018-04-01/examples/iothub_testnewroute.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        "body": "Body of message",
+        "appProperties": "App Properties",
+        "systemProperties": "System Properties"
+      },
+      "route": {
+        "name": "Route id",
+        "source": "DeviceMessages",
+        "endpointNames": [
+          "id1",
+          "id2"
```
Comment (jianghaolu) [inline specification/iothub/resource-manager/Microsoft.Devices/stable/2018-04-01/examples/iothub_testnewroute.json]: Thanks for addressing my comment - please also fix it here.

File: specification/iothub/resource-manager/Microsoft.Devices/stable/2018-04-01/examples/iothub_testallroutes.json [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      }
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "routes": [
+          {
+            "properties": {
+                "name": "Route id",
```
Comment (jianghaolu) [inline specification/iothub/resource-manager/Microsoft.Devices/stable/2018-04-01/examples/iothub_testallroutes.json]: Name doesn't match requirement ^[A-Za-z0-9-._]{1,64}$: https://github.com/Azure/azure-rest-api-specs/pull/3031/files#diff-54dafbe4198f749bac837b913372bcddR2393

File: specification/iothub/resource-manager/Microsoft.Devices/stable/2018-04-01/examples/iothub_testnewroute.json [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    "api-version": "2018-04-01",
+    "subscriptionId": "91d12660-3dec-467a-be2a-213b5544ddc0",
+    "input": {
+      "message": {
+        "body": "Body of message",
+        "appProperties": "App Properties",
+        "systemProperties": "System Properties"
+      },
+      "route": {
+        "name": "Route id",
```
Comment (jianghaolu) [inline specification/iothub/resource-manager/Microsoft.Devices/stable/2018-04-01/examples/iothub_testnewroute.json]: Same here

**Review by jianghaolu — APPROVED** [review]

**Pull request merged**